src/mmrelay/plugin_loader.py

```
@@ -577,32 +619,50 @@ def load_plugins_from_directory(directory, recursive=False):
                         # Try to automatically install the missing dependency
                         try:
+                            auto_install = bool(
+                                (config or {})
+                                .get("security", {})
+                                .get("auto_install_deps", True)
+                            )
+                            if not auto_install:
+                                logger.warning(
+                                    f"Auto-install disabled; cannot install {missing_pkg}. See docs for enabling."
+                                )
+                                raise subprocess.CalledProcessError(1, "pip/pipx")
```

Ruff (0.13.1)

Abstract `raise` to an inner function

(TRY301)